candle-wasm-examples/segment-anything/src/bin/m.rs

```diff
@@ -74,17 +75,31 @@ impl Model {
         Ok(())
     }
 
-    // x and y have to be between 0 and 1
-    pub fn mask_for_point(&self, x: f64, y: f64) -> Result<JsValue, JsError> {
-        if !(0. ..=1.).contains(&x) {
-            Err(JsError::new(&format!(
-                "x has to be between 0 and 1, got {x}"
-            )))?
-        }
-        if !(0. ..=1.).contains(&y) {
-            Err(JsError::new(&format!(
-                "y has to be between 0 and 1, got {y}"
-            )))?
+    pub fn mask_for_point(&self, points: js_sys::Array) -> Result<JsValue, JsError> {
```

Comment:
Hi @lichin-lin instead of using `js_sys` here, we could use the already imported `serde_wasm_bindgen` so it's easy to send data from js to wasm, in addition to offer an opening to sending extra data from the ui, the boolean to set background point or not.

```rust
    pub fn mask_for_point(&self, input: JsValue) -> Result<JsValue, JsError> {
        let input: PointsInput =
            serde_wasm_bindgen::from_value(input).map_err(|m| JsError::new(&m.to_string()))?;
        let transformed_points = input.points;
```
and the implementation

```rust
[derive(serde::Serialize, serde::Deserialize)]
struct PointsInput {
    points: Vec<(f64, f64, bool)>,
}
````